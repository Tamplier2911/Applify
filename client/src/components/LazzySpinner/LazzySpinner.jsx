// import './LazzySpinner.scss';
import React from "react";

// JS Rendering CSS
import {
  LazzySpinnerContainer,
  LazzySpinnerSquare,
} from "./LazzySpinnerStyles";

const LazzySpinner = () => {
  return (
    <LazzySpinnerContainer data-test="lazzy-spinner">
      <LazzySpinnerSquare data-test="lazzy-spinner-square" />
    </LazzySpinnerContainer>
  );
};

export default LazzySpinner;
