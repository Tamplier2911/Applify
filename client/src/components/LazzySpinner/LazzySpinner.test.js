import React from "react";
import toJson from "enzyme-to-json";

// component
import LazzySpinner from "./LazzySpinner";

// utils
import { setShallow, findByTestAttr } from "../../tests/testUtils";

describe("<LazzySpinner />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<LazzySpinner />);
  });

  it("renders without an error", () => {
    const lazzySpinner = findByTestAttr(wrapper, "lazzy-spinner");
    expect(lazzySpinner.length).toBe(1);
  });

  it("renders children - loading square", () => {
    const lazzySpinnerSquare = findByTestAttr(wrapper, "lazzy-spinner-square");
    expect(lazzySpinnerSquare.length).toBe(1);
  });

  it("matches snapshot", () => {
    const lazzySpinner = findByTestAttr(wrapper, "lazzy-spinner");
    expect(toJson(lazzySpinner)).toMatchSnapshot();
  });
});
