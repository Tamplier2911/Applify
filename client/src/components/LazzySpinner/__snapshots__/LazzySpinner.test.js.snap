// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazzySpinner /> matches snapshot 1`] = `
<styled.div
  data-test="lazzy-spinner"
>
  <styled.div
    data-test="lazzy-spinner-square"
  />
</styled.div>
`;
